q42stats/src/main/java/com/q42/q42stats/library/Q42Stats.kt

```diff
@@ -70,12 +70,11 @@ class Q42Stats(private val config: Q42StatsConfig) {
                 val batchId = it.getString("batchId") // throws if not found
                 prefs.lastBatchId = batchId
                 prefs.previousMeasurement = currentMeasurement
-                prefs.updateSubmitTimestamp()
             }
-
         } catch (e: Throwable) {
             handleException(e)
         } finally {
+            prefs.updateSubmitTimestamp() // make sure to always update the submit timestamp
```

Comment:
There is no reference to `prefs` in the finally context. Maybe you can initialize prefs before the try catch block?

Reply:
Wew, my bad. Linter wasn't working properly! I've moved to initialisation of shared prefs to a separate function so we do not have to deal with nested try / catch blocks within the same function